pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -209,14 +238,10 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 		foreign_payout_currency: T::CurrencyId,
 		pool_currency: T::CurrencyId,
 	) -> Result<(), DispatchError> {
+		let payout_currency = RedemptionPayoutCurrency::<T>::get(who, investment_id)?;
```

Comment:
Not sure if the same thinking applies to redemptions.

Reply:
A `RedemptionState` is only cleared if 
1. either the entire unprocessed amount is decreased without an active/done swap (`Redeeming` --> `NoState`) or
2. the entire swap is fulfilled without unprocessed redeeming amount (`SwapIntoForeignDone` --> `NoState`)

The first one is irrelevant here because it requires the `RedemptionState` to be re-initialized by increasing the redemption which sets the payout currency.

The second one triggers the redemption collection transition before/when clearing:

https://github.com/centrifuge/centrifuge-chain/blob/0b6924df3d14d6c18a15f760746259baca6b47c2/pallets/foreign-investments/src/impls/mod.rs#L503-L509

https://github.com/centrifuge/centrifuge-chain/blob/0b6924df3d14d6c18a15f760746259baca6b47c2/pallets/foreign-investments/src/impls/mod.rs#L706-L714